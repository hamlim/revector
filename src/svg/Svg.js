import React from 'react'

export const context = React.createContext({
  dimensions: {
    height: 0,
    width: 0,
  },
})

export class Svg extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      dimensions: {
        height: props.height,
        width: props.width,
      },
      update: fn => this.setState(fn),
    }
  }
  render() {
    const { height, width, ...rest } = this.props
    return (
      <context.Provider value={this.state}>
        <svg {...rest} height={height} width={width} />
      </context.Provider>
    )
  }
}
