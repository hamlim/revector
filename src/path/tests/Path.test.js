import React from 'react'
import { render } from 'react-testing-library'
import { Svg } from '../../svg'
import {
  Path,
  Line,
  Move,
  CubicBezier,
  CubicBezierStart as CBStart,
  CubicBezierControl as CBControl,
  CubicBezierEnd as CBEnd,
} from '../Path'

describe('Path', () => {
  it('matches a snapshot using Move and Line', () => {
    const { container } = render(
      <Svg height="100%" width="100%" viewBox="0 0 200 100">
        <Path fill="none" stroke="red">
          <Move x={10} y={90} />
          <Line x={({ x }) => x + 10} y={({ y }) => y + 10} />
        </Path>
      </Svg>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('matches a snapshot using Move and Cubic Bezier', () => {
    const { container } = render(
      <Svg height={'100%'} width={'100%'} viewBox="0 0 200 100">
        <Path fill="none" stroke="red">
          <Move x={10} y={90} />
          <CubicBezier>
            <CBStart x={30} y={90} />
            <CBControl x={25} y={10} />
            <CBEnd x={50} y={10} />
          </CubicBezier>
          <CubicBezier smooth>
            <CBStart x={70} y={90} />
            <CBEnd x={90} y={90} />
          </CubicBezier>
        </Path>
      </Svg>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
