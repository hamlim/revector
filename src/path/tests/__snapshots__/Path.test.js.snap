// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Path matches a snapshot using Move and Cubic Bezier 1`] = `
<svg
  viewBox="0 0 200 100"
  height="100%"
  width="100%"
>
  <path
    d=" M 10, 90 C 30,90 25,10 50,10 S 70,90  90,90"
    fill="none"
    stroke="red"
  />
</svg>
`;

exports[`Path matches a snapshot using Move and Line 1`] = `
<svg
  viewBox="0 0 200 100"
  height="100%"
  width="100%"
>
  <path
    d=" M 10, 90 L 20, 100"
    fill="none"
    stroke="red"
  />
</svg>
`;
